spec/services/cfe/create_applicant_service_spec.rb

```diff
@@ -1,30 +1,63 @@
 require 'rails_helper'
 
-module CFE # rubocop:disable Metrics/ModuleLength
+module CFE
   RSpec.describe CreateApplicantService do
-    # uncomment this if you need to connect to a real instance of Check Financial Eligibility service
-    # running on localhost
-    #
-    # before(:all) do
-    #   VCR.configure { |c| c.ignore_hosts 'localhost' }
-    # end
-    #
-    let(:application) { create :legal_aid_application, application_ref: 'L-XYZ-999' }
-    let!(:applicant) { create :applicant, legal_aid_application: application, date_of_birth: Date.new(1999, 9, 11) }
+    let(:application) { create :legal_aid_application }
+    let!(:applicant) { create :applicant, legal_aid_application: application }
     let(:submission) { create :cfe_submission, aasm_state: 'assessment_created', legal_aid_application: application }
-    let(:faraday_connection) { double Faraday }
-    let(:connection_param) { double.as_null_object }
-    let(:expected_payload) do
+    let(:expected_payload) { expected_payload_hash.to_json }
+    let(:expected_response) { expected_response_hash.to_json }
+    let(:cfe_host) { Rails.configuration.x.check_finanical_eligibility_host }
+    let(:cfe_url) { "#{cfe_host}/assessments/#{submission.assessment_id}/applicant" }
+
+    context 'response received from CFE' do
+      before do
+        allow(application).to receive(:applicant_receives_benefit?).and_return(true)
+        allow(application).to receive(:calculation_date).and_return(Date.today)
+        allow_any_instance_of(described_class).to receive(:request_body).and_return(expected_payload)
```

Comment:
I strongly believe that we should only be mocking when there is no alternative.

- The only relevance of `applicant_receives_benefit?` in this spec is that the entry for the application matches that in the payload. So the default `false` is just as valid and the mocked true - so no need to mock.
- If `transaction_period_finish_on` is set on the application, that will be used as the `calculation_date` so that can be done in the factory call, without having to mock.
- `request_body` is the fundamental property of this service. As `expected_payload` gets passed to both the instance of the service and the webmock declaration ALL THAT IS BEING TESTED IS THAT MOCKING IS WORKING. Nowhere is the construction of `request_body` tested, and that is the most important part of the spec!!!!!